### ISSUE TITLE
Failed to legalize operation 'stablehlo.batch_norm_training'

### ISSUE THREAD

**Issue by mrakitaTT** [tenstorrent/tt-xla#735]
PyTorch test for albert_v2 model fails with:
```
error: failed to legalize operation 'stablehlo.batch_norm_training'
```

We should investigate how do we even end up with `batch_norm_training` op in the inference test, and also add support for this op in compiler.

**Comment by milank94** [comment]
Thanks for catching that, @mrakitaTT. Yes, I think it was due to the new IT SSO changes -- we've seen this in other repos.

**Comment by vkovacevicTT** [comment]
@mrakitaTT We have the same issue when running ViT and Segformer through torch-xla in tt-forge

**Comment by nvukobratTT** [comment]
@vkovacevicTT , QQ, did you try setting something like `framework_model = framework_model.eval()`? Batch norm train probably comes from BN op when model is set in training mode. More precisely, even forward graph can differ when model is in ether inference or training mode (batch norm is a good example) :))

**Comment by vkovacevicTT** [comment]
We have framework.eval() for every model.
Here is an example: https://github.com/tenstorrent/tt-forge/blob/07f5c901515fa1d510c3dd1c5b6a3dfa03ebee1c/benchmark/tt-xla/mobilenetv2.py#L109

**Comment by kmabeeTT** [comment]
Yeah, can confirm all albert tests in tt-forge-models repo pass in tt-xla with upcoming changes (tt backend via torch_xla).. didn't see above error anywhere.

```
grep albert tests/runner/test_config.py
    "albert/masked_lm/pytorch-xxlarge_v2-full-inference": {
    "albert/masked_lm/pytorch-large_v2-full-inference": {
    "albert/masked_lm/pytorch-base_v2-full-inference": {
    "albert/masked_lm/pytorch-xlarge_v2-full-inference": {
    "albert/token_classification/pytorch-base_v1-full-inference": {
    "albert/token_classification/pytorch-large_v1-full-inference": {
    "albert/token_classification/pytorch-xxlarge_v1-full-inference": {
    "albert/masked_lm/pytorch-base_v1-full-inference": {
    "albert/masked_lm/pytorch-large_v1-full-inference": {
    "albert/masked_lm/pytorch-xlarge_v1-full-inference": {
    "albert/masked_lm/pytorch-xxlarge_v1-full-inference": {
    "albert/question_answering/pytorch-squad2-full-inference": {
    "albert/sequence_classification/pytorch-imdb-full-inference": {
    "albert/token_classification/pytorch-large_v2-full-inference": {
    "albert/token_classification/pytorch-xlarge_v1-full-inference": {
    "albert/token_classification/pytorch-base_v2-full-inference": {
    "albert/token_classification/pytorch-xxlarge_v2-full-inference": {
    "albert/token_classification/pytorch-xlarge_v2-full-inference": {
```

**Comment by vkovacevicTT** [comment]
UPDATE: setting backend to `tt` (`framework_model.compile(backend="tt")`) seems to have solved the problem for segformer and vit
